tools/SourceKit/tools/sourcekitd/lib/API/sourcekitdAPI-XPC.cpp

```diff
@@ -379,8 +379,9 @@ sourcekitd::createErrorRequestInvalid(const char *Description) {
   return CustomXPCData::createErrorRequestInvalid(Description).getXObj();
 }
 sourcekitd_response_t
-sourcekitd::createErrorRequestFailed(const char *Description) {
-  return CustomXPCData::createErrorRequestFailed(Description).getXObj();
+sourcekitd::createErrorRequestFailed(StringRef DescRef) {
+  std::string Desc = DescRef.str();
+  return CustomXPCData::createErrorRequestFailed(Desc.c_str()).getXObj();
```

Comment:
Could you change `CustomXPCData::createErrorRequestFailed` to accept a `StringRef` instead ?
Also for consistency please change the other functions as well:
* `sourcekitd::createErrorRequestInvalid`
* `sourcekitd::createErrorRequestInterrupted`
* `CustomXPCData::createErrorRequestInvalid`
* `CustomXPCData::createErrorRequestInterrupted`
* `CustomXPCData::createErrorRequestCancelled`

The `CustomXPCData` functions are just calling`createKindAndString`, change that to accept a `StringRef` like this:
```
  static CustomXPCData createKindAndString(Kind K, StringRef Str) {
    llvm::SmallVector<char, 128> Buf;
    Buf.push_back((char)K);
    Buf.append(Str.begin(), Str.end());
    Buf.push_back('\0');
```

Reply:
Done